host/class/uvc/usb_host_uvc_2/private_include/uvc_types_priv.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <inttypes.h>
+#include <stdbool.h>
+#include <sys/queue.h>
+
+#include "usb/usb_host.h"
+#include "usb/uvc_host.h"
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/queue.h"
+
+typedef struct uvc_host_stream_s uvc_stream_t;
+
+/**
+ * @brief Enum for simple state machine of Bulk frame data processing
+ */
+typedef enum {
+    UVC_STREAM_BULK_PACKET_SOF = 0,
+    UVC_STREAM_BULK_PACKET_DATA,
+    UVC_STREAM_BULK_PACKET_EOF,
+} uvc_stream_bulk_packet_type_t;
+
+struct uvc_host_stream_s {
+    // UVC driver related members
+    uvc_host_stream_callback_t stream_cb; // User's callback for stream events
+    uvc_host_frame_callback_t frame_cb;   // User's frame callback
+    void *cb_arg;                         // Common argument for user's callbacks
+    SLIST_ENTRY(uvc_host_stream_s) list_entry;
+
+    // Constant USB descriptor values
+    uint16_t bcdUVC;                      // Version of UVC specs this device implements
+    uint8_t bInterfaceNumber;             // USB Video Streaming interface claimed by this stream. Needed for ISOC Stream start and CTRL transfers
+    uint8_t bAlternateSetting;            // Alternate setting for selected interface. Needed for ISOC Stream start
+    uint8_t bEndpointAddress;             // Streaming endpoint address. Needed for BULK Stream stop
+
+    // USB host related members
+    usb_device_handle_t dev_hdl;          // USB device handle
+    unsigned num_of_xfers;                // Number of USB transfers
+    usb_transfer_t **xfers;               // Pointer to array of USB transfers. Accessible only by the UVC driver
+
+    // Frame buffers
+    bool streaming;                       // Flag whether the stream is on/off
+    uvc_host_stream_format_t vs_format;   // Format of the video stream
+    QueueHandle_t empty_fb_queue;         // Queue of empty framebuffers
+    uvc_host_frame_t *current_frame;      // Frame that is being written to
+    bool skip_current_frame;              // Flag to skip this frame. An error has occurred during fetch
+    uint8_t current_frame_id;             // Current frame ID. Used for start of frame detection
```

Comment:
seems, that streaming, skip_current_frame and current_frame_id could be implemented in one member, something like: 

```
dynamic.flags.streaming (1 bit)
dynamic.flags.skip_current_frame (1 bit)
dynamic.flags.current_frame_id (8 bit)
```

Reply:
Now these variables were split to `dynamic` and `single_thread` sub structs, so this optimization loses some of its effeciency.

Overall I'm not a big fan of this 'flags optimization' because the compiler can optimize this for me, so the speed and memory requirements are the same...

Reply:
Sure, but I meant two main point in my comment: 
1. seems, that all flags are dynamic members
2. optimized size (which usually not that critical, in case the potential profit is just couple of bytes

If only one of points is applied, than it ok )